File: Frontend/sakai-ng-master/src/app/biblioteca/interfaces/programa-accion.ts
export class ProgramaAccion {
    id: number;
    descripcion: string;
    estado: boolean;
    constructor(init?: Partial<ProgramaAccion>) {
        this.id = 0;
        this.descripcion = '';
        this.estado = true;
        Object.assign(this, init);
    }
}

File: Backend/login-microsoft365/src/main/java/com/miapp/model/ProgramaAccion.java
package com.miapp.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "PROGRAMA_ACCION")
public class ProgramaAccion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "IDPROGRAMAACCION")
    private Long idProgramaAccion;

    @Column(name = "DESCRIPCION", length = 100, nullable = false)
    private String descripcion;

    @Column(name = "ESTADO", columnDefinition = "NUMBER(1)")
    private Boolean estado = true;
}